python/ray/_private/metrics_agent.py

```diff
@@ -7,27 +7,18 @@
 from collections import namedtuple
 from typing import List
 
-from ray import logger
-
-try:
-    from opencensus.stats import aggregation
-    from opencensus.stats import measure as measure_module
-    from opencensus.stats import stats as stats_module
-    from opencensus.stats.view import View
-    from opencensus.stats.view_data import ViewData
-    from opencensus.stats.aggregation_data import (CountAggregationData,
-                                                   DistributionAggregationData,
-                                                   LastValueAggregationData)
-    from opencensus.metrics.export.value import ValueDouble
-    from opencensus.tags import tag_key as tag_key_module
-    from opencensus.tags import tag_map as tag_map_module
-    from opencensus.tags import tag_value as tag_value_module
-except (ModuleNotFoundError, ImportError):
-    gpustat = None
-    logger.warning("`gpustat` package is not installed. GPU monitoring is "
-                   "not available. In Ray 1.4+, the Ray CLI, autoscaler, and "
-                   "dashboard will only be usable via `pip install 'ray["
-                   "default]'`. Please update your install command")
+from opencensus.stats import aggregation
+from opencensus.stats import measure as measure_module
+from opencensus.stats import stats as stats_module
+from opencensus.stats.view import View
+from opencensus.stats.view_data import ViewData
+from opencensus.stats.aggregation_data import (CountAggregationData,
+                                               DistributionAggregationData,
+                                               LastValueAggregationData)
+from opencensus.metrics.export.value import ValueDouble
+from opencensus.tags import tag_key as tag_key_module
+from opencensus.tags import tag_map as tag_map_module
+from opencensus.tags import tag_value as tag_value_module
 
```

Comment:
Hmm, doesn't this cause us to continue depending on opencensus?

Reply:
yeah `metrics_agent` does depend on `opencensus`, and I think it should. It's just that if we don't have this installed, then we shouldn't start the dashboard in the first place and this is covered by  https://github.com/ray-project/ray/pull/17420/files#diff-8607e53e10994032eb1356039356d1342d131d98631a4d552c418fd55c270cbdR1178 and https://github.com/ray-project/ray/pull/17420/files#diff-85bf4cf80807eedf282aabf780b27cb0ea7216d0b0f9386b3f0afcf0301964bcR4.

Reply:
but cc @simon-mo just to confirm. Is `opencensus` a strict requirement for the dashboard? what about the other dependencies?